sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs

```diff
@@ -364,6 +338,13 @@ public void AddLink(string traceparent, string? tracestate, IDictionary<string,
                     links: GetActivitySourceLinkCollection(),
                     traceparent: _traceparent,
                     tracestate: _tracestate);
+
+                if (activity == null || ((bool?)ActivityExtensions.GetCustomProperty(activity, "IsAllDataRequested") ?? false))
```

Comment:
`IsAllDataRequested` is a flag on the activity.

`GetCustomProperty` method calls into [Activity.GetCustomProperty](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs#L1046), what you need is to call [Activity.IsAllDataRequested](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs#L837).

You'd have to add a new method that gets `IsAllDataRequested` - you can use `ActivitySourceHasListeners` as an example on how to do it with reflection+caching for property